packages/slate/src/controllers/editor.js

```diff
@@ -100,14 +100,12 @@ class Editor {
 
     // Get the paths of the affected nodes, and mark them as dirty.
     const newDirtyPaths = getDirtyPaths(operation)
-    const dirty = this.tmp.dirty.reduce((memo, path) => {
-      path = PathUtils.create(path)
-      const transformed = PathUtils.transform(path, operation)
-      memo = memo.concat(transformed.toArray())
-      return memo
-    }, newDirtyPaths)
-
-    this.tmp.dirty = dirty
+
+    const dirty = this.tmp.dirty.map(path =>
+      PathUtils.transform(PathUtils.create(path), operation).toArray()
```

Comment:
Is there any reason for collapsing this into a one-liner? Could you re-expand it to be multiple lines like it was previously? Having one-liners like this makes it much hard to debug I've found, since setting breakpoints and checking values is much harder.

Reply:
For sure!